### PR TITLE
fix for mantis bug 0020168

### PR DESCRIPTION
**Pull request by bseglias**
https://www.ilias.de/mantis/view.php?id=20168

### CODE REVIEW

File: Services/Repository/classes/class.ilRepUtil.php [hunk 1]
```diff
@@ -141,6 +141,7 @@ static public function deleteObjects($a_cur_ref_id, $a_ids)
 			// SAVE SUBTREE AND DELETE SUBTREE FROM TREE
 			$affected_ids = array();
 			$affected_parents = array();
+			$org_unit_ids = array();
```
Comment (chfsx) [inline Services/Repository/classes/class.ilRepUtil.php]: @bseglias could you remove the changes from ilRepUtil and ilRepUtilGUI and override `confirmedDeleteObject` from `ilObjectGUI` in `ilObjOrgUnitGUI` to just remove the orgus directly by using `$ru->removeObjectsFromSystem($_GET["ref_id"], ilSession::get("saved_post"));`? 

`ilRepUtil::showDeleteConfirmation()` then need something like an additional parameter `$additional_message` defaulted to null.

File: Services/Repository/classes/class.ilRepUtil.php [hunk 2]
```diff
@@ -188,6 +189,11 @@ static public function deleteObjects($a_cur_ref_id, $a_ids)
 
 				// TODO: inform users by mail that object $id was deleted
 				//$mail->sendMail($id,$msg,$affected_users);
+
```
Comment (chfsx) [inline Services/Repository/classes/class.ilRepUtil.php]: This class should not know about orgus.

File: Services/Repository/classes/class.ilRepUtil.php [hunk 3]
```diff
@@ -200,11 +206,14 @@ static public function deleteObjects($a_cur_ref_id, $a_ids)
 						 ));
 			}
 		}
-		
 		if (!$ilSetting->get('enable_trash'))
 		{
 			ilRepUtil::removeObjectsFromSystem($a_ids);
 		}
+		if($ilSetting->get('enable_trash') && !empty($org_unit_ids)) {
```
Comment (chfsx) [inline Services/Repository/classes/class.ilRepUtil.php]: This class should not know about orgus.

File: Services/Repository/classes/class.ilRepUtilGUI.php [hunk 4]
```diff
@@ -119,6 +119,11 @@ function showDeleteConfirmation($a_ids, $a_supress_message = false)
 			$alt = ($objDefinition->isPlugin($type))
 				? $lng->txt("icon")." ".ilObjectPlugin::lookupTxtById($type, "obj_".$type)
 				: $lng->txt("icon")." ".$lng->txt("obj_".$type);
+
+			if($type == "orgu" && isset($msg)) {
```
Comment (chfsx) [inline Services/Repository/classes/class.ilRepUtilGUI.php]: This class should not know about orgus.

File: lang/ilias_de.lang [hunk 5]
```diff
@@ -2944,6 +2944,7 @@ common#:#info_deactivate_sure#:#Sind Sie sicher, dass Sie die folgenden Benutzer
 common#:#info_delete_sure#:#Sind Sie sicher, dass folgende Objekte gelöscht werden sollen:
 common#:#info_remove_sure#:#Sind Sie sicher, dass folgende Objekte entfernt werden sollen:
 common#:#info_delete_warning_no_trash#:#(WARNUNG: Die ausgewählten Objekte werden unwiderruflich aus dem System entfernt und können nicht wiederhergestellt werden!)
+common#:#info_delete_warning_no_trash_for_org_units#:# (WARNUNG: Organisationseinheiten werden unwiderruflich aus dem System entfernt und können nicht wiederhergestellt werden!)
```
Comment (chfsx) [inline lang/ilias_de.lang:2949]: Thanks, please add the english translation as well

File: setup/sql/5_3_hotfixes.php [hunk 6]
```diff
@@ -106,4 +106,20 @@
...
+//removes org units from trash
+global $DIC;
+$obj_set = $DIC->database()->queryF('SELECT * FROM tree JOIN object_reference 
+ON tree.child=object_reference.ref_id JOIN object_data 
+ON object_reference.obj_id=object_data.obj_id 
+WHERE tree.tree < %s AND object_data.type = %s',array('integer', 'text'),array(0, "orgu"));
+$a_org_unit_ref_ids = array();
+while($row = $DIC->database()->fetchAssoc($obj_set))
+{
+	$a_org_unit_ref_ids[] = $row;
```
Comment (chfsx) [inline setup/sql/5_3_hotfixes.php]: `$row` is an array I think, therefore you should collect something like `$a_org_unit_ref_ids[] = $row['child']`. Please retest after.

**Review by chfsx — CHANGES_REQUESTED** [review]

**Comment by chfsx** [comment]
Additionally to the commets above there seems to be a conflict in `setup/sql/5_3_hotfixes.php`. Please merge latest release_5-3 to you branch an resolve the conflicts. The new hotfix-step should be the latest then.

File: setup/sql/5_3_hotfixes.php [hunk 7]
```diff
@@ -106,4 +106,20 @@
...
+}
+ilRepUtil::removeObjectsFromSystem($a_org_unit_ref_ids);
```
Comment (chfsx) [inline setup/sql/5_3_hotfixes.php]: @smeyer-ilias Is it ok the use `ilRepUtil`in db_update in general? Was not sure... Or should we introduce a copy of the class in `Services/Migration`? Thanks for the feedback!

File: setup/sql/5_3_hotfixes.php [hunk 8]
```diff
@@ -106,4 +106,20 @@
 <#7>
 <?php
 $ilCtrlStructureReader->getStructure();
-?>
\ No newline at end of file
+?>
+<#8>
+<?php
+//removes org units from trash
```
Comment (chfsx) [inline setup/sql/5_3_hotfixes.php]: @bseglias could you add a link to the mantis-ticket 0020168 here in this comment? Thanks!

File: lang/ilias_de.lang (hunk 5 above)
Comment (bseglias) [reply lang/ilias_de.lang:2949]: @chfsx The english translation exists on my local bug fix branch.

**Comment by chfsx** [comment]
@bseglias could you please open an internal MR on our gitlab for trunk instead of release_5-3? Thanks in advance!

**Pull request closed without merging**